SwiftCompilerSources/Sources/Optimizer/ModulePasses/MandatoryPerformanceOptimizations.swift

```diff
@@ -146,6 +181,20 @@ fileprivate struct FunctionWorklist {
     }
   }
 
+  mutating func addAllAnnotatedGlobalInitOnceFunctions(of moduleContext: ModulePassContext) {
+    for f in moduleContext.functions where f.isGlobalInitOnceFunction {
+      guard let global = getGlobal(of: f) else {
```

Comment:
nitpick: Instead of multiple guard statements you can make a single if statement, combining all the conditions, e.g.
```
  if let global = f. getInitializedGlobal(),
     global.mustBeInitializedStatically {
    pushIfNotVisited(f)
  }
```
IMO, this reads more naturally.